src/test/scala/cromwell/engine/db/slick/SlickDataAccessSpec.scala

```diff
@@ -213,6 +221,122 @@ class SlickDataAccessSpec extends FlatSpec with Matchers with ScalaFutures {
       } yield ()).futureValue
     }
 
+    it should "support call caching configuration for specified calls" in {
+      assume(canConnect || testRequired)
+
+      def assertFailureFor(id: WorkflowId, callName: String, messages: String*): Future[Unit] = {
+        // The `from` Future is expected to fail, so if the `map` block runs the test should fail.  `recover` a failed
+        // Future and assert all the expected messages are present in the exception text and the correct number of
+        // expected failures are seen.
+        CallCachingParameters.from(id, Option(callName), AllowFalse, dataAccess) map {
+          s => throw new RuntimeException(s"Unexpected success: $s") } recover {
+          case e: IllegalArgumentException =>
+            messages foreach { m => if (!e.getMessage.contains(m)) throw new RuntimeException(s"Missing message: $m.  Exception text: ${e.getMessage}") }
+            if (e.getMessage.count(_ == '\n') != messages.size - 1) throw new RuntimeException(s"Unexpected messages seen: ${e.getMessage}")
+        }
+      }
```

Comment:
I'm still going through this block, but I believe this could be simplified with `val exception = myFuture.failed.futureValue`, and then further processing the expected exception? One would loose some of the specificity in the error when something goes wrong in the test, but nothing will go wrong. Ever.